sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -238,13 +241,25 @@ def get_range_tracker(self, start_position, stop_position):
       tracker = range_trackers.UnsplittableRangeTracker(tracker)
     return tracker
 
+  def _gen_kv_pair(self, index):
+    r = np.random.RandomState(index)
+    rand = r.random_sample()
+
+    # Determines whether to generate hot key or not.
+    if rand < self._hot_key_fraction:
+      # Generate hot key.
+      # An integer is randomly selected from the range [0, numHotKeys-1]
+      # with equal probability.
+      r_hot = np.random.RandomState(self._num_hot_keys)
+      return r_hot.bytes(self._key_size), r.bytes(self._value_size)
```

Comment:
@pabloem From your previous comment, I started thinking there is also bug here. In [Java](https://github.com/apache/beam/blob/8373a57525abd88fd476c70b73dc7ac3615499f7/sdks/java/io/synthetic/src/main/java/org/apache/beam/sdk/io/synthetic/SyntheticOptions.java#L346) the upper bound is a `hot_key_number` and I'm not sure if it's ensured also here.